ci/iothub-get-twin.sh

```diff
@@ -0,0 +1,297 @@
+            ($packet_type, $packet_flags, $packet_payload) = read_packet();
+            if ($packet_type != 0x20 || $packet_flags != 0) {
+                die "expected CONNACK but got $packet_type:$packet_flags";
+            }
+            if ($packet_payload ne "\x00\x00") {
+                die "connection failed";
+            }
+
+            # SUBACK { packet_identifier: 1, qos: [AtMostOnce] }
+            ($packet_type, $packet_flags, $packet_payload) = read_packet();
+            if ($packet_type != 0x90 || $packet_flags != 0) {
+                die "expected SUBACK but got $packet_type:$packet_flags";
+            }
+            if ($packet_payload ne "\x00\x01\x00") {
+                die "subscription failed";
+            }
+
+            # PUBLISH { dup: false, retain: false, qos: AtMostOnce, topic_name: "$iothub/twin/res/200/?$rid=1", payload }
+            ($packet_type, $packet_flags, $packet_payload) = read_packet();
+            if ($packet_type != 0x30 || $packet_flags != 0) {
+                die "expected PUBLISH but got $packet_type:$packet_flags";
+            }
+            my $topic_length = unpack("n", substr($packet_payload, 0, 2));
+            my $topic = substr($packet_payload, 2, $topic_length);
+            if ($topic ne "\$iothub/twin/res/200/?\$rid=0") {
+                die "expected PUBLISH with topic \"\$iothub/twin/res/200/?\$rid=0\" but got PUBLISH with topic \"$topic\"";
+            }
+            print substr($packet_payload, 2 + $topic_length);
+
+
+            # DISCONNECT
+            write_packet($fh, "\xE0", "");
+        ' -- \
+            "/proc/$BASHPID/fd/${CONNECTION[1]}" \
+            "$client_id" \
+            "$username" \
+            "$password"
+    ) | jq '.'
+}
+
+
+identity="$1"
+
+hub_name="$(<<< "$identity" jq -r '.spec.hubName')"
+device_id="$(<<< "$identity" jq -r '.spec.deviceId')"
+module_id="$(<<< "$identity" jq -r '.spec.moduleId // ""')"
+auth_type="$(<<< "$identity" jq -r '.spec.auth.type')"
+key_handle="$(<<< "$identity" jq -r '.spec.auth.keyHandle')"
+
+if [ -z "$module_id" ]; then
+    resource_uri="$(printf '%s/devices/%s' "$hub_name" "$device_id" | sed -e 's|/|%2f|g')"
+    client_id="$device_id"
+    username="$hub_name/$device_id/?api-version=2018-06-30"
+else
+    resource_uri="$(printf '%s/devices/%s/modules/%s' "$hub_name" "$device_id" "$module_id" | sed -e 's|/|%2f|g')"
+    client_id="$device_id/$module_id"
+    username="$hub_name/$device_id/$module_id/?api-version=2018-06-30"
+fi
+
+case "$auth_type" in
+    'sas')
+        if [ -z "$module_id" ]; then
+            # TODO: Bug? For device ID spec, the key handle is actually the key ID.
+            key_id="$(uri_encode "$key_handle")"
+            key_handle="$(
+                curl --unix-socket '/run/aziot/keyd.sock' \
+                    "http://foo/key/$key_id?api-version=2020-09-01" |
+                    jq '.keyHandle' -er
+            )"
```

Comment:
@massand Is this expected?

Reply:
Bug, I'll fix it in a PR today (and fix the test if this goes in earlier).

Reply:
Thanks.